### PR TITLE
Adding System.setContextual

### PR DESCRIPTION
**Pull request by phillipskevin**
Closes https://github.com/stealjs/steal/issues/518.

### CODE REVIEW

**Comment by matthewp** [comment]
I think you had a different version of some upstream projects than what is is minor, which is why there are all of the unrelated changes showing up. If you rm -rf node_modules bower_components and reinstall everything, do another build, should fix it I think/hope.

**Comment by phillipskevin** [comment]
@matthewp Thanks. I didn't realize that's what was causing those changes. Should be good to go now.

File: main.js [hunk 1]
```diff
@@ -392,6 +392,51 @@ if(typeof System !== "undefined") {
 	addTilde(System);
 }
 
+function addContextual(loader){
+  loader._contextualModules = {};
+
+  loader.setContextual = function(moduleName, definer){
+    // `definer` could be a function or could be a moduleName
+		if (typeof definer === 'string') {
+			this._contextualModules[moduleName] = this['import'](definer);
```
Comment (matthewp) [inline main.js]: Like we talked about yesterday, doing the import here means that the import happens when `setContextual` is called, even if no one is going to be using this module. We should just save the string name here and import it in `normalize` when it's needed.

Comment (phillipskevin) [inline main.js]: I was going to put that on the steal-clone PR. Do you want it on this branch instead?

Comment (matthewp) [inline main.js]: Yeah, this one should be, I wouldn't want to merge this without that fix.

**Comment by phillipskevin** [comment]
@matthewp I've updated the PR to only import contextual modules when they are used if you want to take another look.

**Comment by phillipskevin** [comment]
@matthewp any issue with merging this?

**Comment by matthewp** [comment]
Nope, looks good.

**Pull request merged**